### PR TITLE
[W5.4][W09-A2]Labayna Neil Brian Narido

### PR DESCRIPTION
**Pull request by nbriannl**
Implement Find by Tag Command: findtag

updated user guide
updated existing tests and add new tests, for both JUnit tests and I/O tests

### CODE REVIEW

**Comment by jeffreygohkw** [comment]
Nice implementation! I think one other way to improve this would be to allow finding people who have no tags as well if you were to use FindTag without any other tag attached to it.

**Comment by dalessr** [comment]
The function implemented is good and clear. Perhaps can think about what else can be improved? For example, finding command can make it case insensitive so that users can find persons more easily.

File: doc/UserGuide.md [hunk 1]
```diff
@@ -66,6 +66,19 @@ Examples:
 * `find Betsy Tim John`<br>
   Returns Any person having names `Betsy`, `Tim`, or `John`
 
+### Finding all persons containing give tags : `findtag`
```
Comment (damithc) [inline doc/UserGuide.md]: give -> given?

File: src/seedu/addressbook/commands/FindTagCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
+        for(String tag: tags) {
+            for (ReadOnlyPerson person : addressBook.getAllPersons()) {
+                for (Tag s : person.getTags()) {
+                    boolean doesTagMatches = s.getTagName().equals(tag);
+                    boolean isDuplicate = matchedPersons.contains(person);
+                    if (doesTagMatches && !isDuplicate){
+                        matchedPersons.add(person);
+                        break;
+                    }
```
Comment (damithc) [inline src/seedu/addressbook/commands/FindTagCommand.java]: If you use a `Set`, you don't have to worry about duplicates.

**Pull request closed without merging**